### ISSUE TITLE
release 4.5.0 breaks pydoc-markdown[novella]

### ISSUE THREAD

**Issue by mgor** [NiklasRosenstein/python-databind#64]
With the release of 4.5.0, and specifically:

> Merged databind.core and databind.json packages into databind. The old PyPI packages will remain as proxies until the next minor version.

`novella` fails to build documentation with a `pkg_resource.DistributionNotFound` exception complaining that `databind.json` isn't found, which is required by `pydoc_markdown.novella.preprocessors.PydocTagPreprocessor`.

I'm guessing this is due to the dummy/proxy package that `databind.json` provides after the merge.

Was able to get around the problem by pinning the version to `4.4.2`.

**Comment by NiklasRosenstein** [comment]
After some more rigorous investigation I found that this seems to be caused not by the proxy packages as originally assumed also in https://github.com/NiklasRosenstein/pydoc-markdown/issues/323, but by the fact that the distribution name's dot (`.`) is normalized to a hyphen (`-`), which `pkg_resources` fails to look up as the `Requires-Dist` of `pydoc-markdown` still references it with a dot (i.e. `databind.json`).

```
$ python3 -m venv venv
$ . venv/bin/activate
$ pip install pydoc-markdown databind.json==4.5.0
$ python -c 'import pkg_resources; pkg_resources.require("databind.json")'; echo $?
pkg_resources.DistributionNotFound: The 'databind.json' distribution was not found and is required by the application
1
$ python -c 'import pkg_resources; pkg_resources.require("databind-json")'; echo $?
0
```

**Comment by NiklasRosenstein** [comment]
The issue is caused by the Poetry update in #60

https://github.com/NiklasRosenstein/python-databind/pull/60/files#diff-ab2ed3985fafcc34e9fb63daaf75dd4417f584b0a339ac6d834a1f476139ac93R16

**Comment by NiklasRosenstein** [comment]
I think this is fixed now with `4.5.1` (#65)

```
$ python3 -m venv venv
$ . venv/bin/activate
$ pip install pydoc-markdown databind.json==4.5.1
$ python -c 'import pkg_resources; pkg_resources.require("databind.json")'; echo $?
0
```

**Comment by mgor** [comment]
Confirmed that 4.5.1 works, thanks!

```python
vscode ➜ /workspaces/.../docs (main) $ export DATABIND_VERSION=4.5.0; python -m pip install databind=="$DATABIND_VERSION" databind.core=="$DATABIND_VERSION" databind.json=="$DATABIND_VERSION" &> /dev/null; python -m novella &> /dev/null || echo "failed"
failed
vscode ➜ /workspaces/.../docs (main) $ export DATABIND_VERSION=4.5.1; python -m pip install databind=="$DATABIND_VERSION" databind.core=="$DATABIND_VERSION" databind.json=="$DATABIND_VERSION" &> /dev/null; python -m novella &> /dev/null || echo "failed"
vscode ➜ /workspaces/.../docs (main) $ 
```
